dice/main.py

```diff
@@ -0,0 +1,103 @@
+from datetime import timedelta
+from json import dumps, loads
+from random import shuffle
+
+from flask import Flask, Response, abort, render_template, request
+
+from common.course_config import get_endpoint
+from common.db import connect_db
+from common.secrets import new_secret
+from contest_utils.oauth import get_group
+from contest_utils.rate_limiting import ratelimited
+
+NUM_DICE = 6
+MAX_CAPTION_LEN = 100
+ASSIGNMENT = "proj01showcase"
+
+app = Flask(__name__, static_folder="", static_url_path="")
+if __name__ == "__main__":
+    app.debug = True
+
+
+with connect_db() as db:
+    db("CREATE TABLE IF NOT EXISTS accesses (email VARCHAR(128), last_access INTEGER)")
+    db(
+        """CREATE TABLE IF NOT EXISTS designs (
+    id varchar(128),
+    email varchar(128),
+    caption varchar(512),
+    dice LONGBLOB,
+    endpoint varchar(512)
+);"""
+    )
+
+
+@app.route("/")
+def index():
+    with connect_db() as db:
+        artworks = db(
+            "SELECT id, caption FROM designs WHERE endpoint=(%s)",
+            [get_endpoint("cs61a")],
+        ).fetchall()
+    shuffle(artworks)
```

Comment:
Ah I see you're shuffling. Shuffling is fun too. Maybe add the timestamp in case we want a sort by time later?